lib/tbot/workloadidentity/workloadattest/podman/client.go

```diff
@@ -0,0 +1,164 @@
+// of the URL path, so setting it doesn't have much of an effect.
+//
+// https://github.com/containers/podman/blob/62cde17193e1469de15995ba78bd909fd07dffc6/pkg/api/server/handler_api.go#L69-L74
+const APIVersion = "v4.0.0"
+
+// Client for Podman's HTTP REST API.
+type Client struct{ httpClient *http.Client }
+
+// NewClient creates a Client for the API service at the given address.
+//
+// Note: addr must be in the form `unix://path/to/socket`, we do not currently
+// support connecting to the Podman service over TCP because it's generally
+// insecure.
+func NewClient(addr string) (*Client, error) {
+	u, err := url.Parse(addr)
+	if err != nil {
+		return nil, trace.Wrap(err, "invalid address: %q", addr)
+	}
+	if u.Scheme != "unix" {
+		return nil, trace.BadParameter("only unix domain sockets are supported")
+	}
+	return &Client{
+		httpClient: &http.Client{
+			Transport: &http.Transport{
+				DialContext: func(ctx context.Context, _ string, _ string) (net.Conn, error) {
+					return (&net.Dialer{}).DialContext(ctx, "unix", u.Path)
+				},
+			},
+		},
+	}, nil
+}
+
+// InspectContainer reads information about the container with the given ID.
+//
+// https://docs.podman.io/en/latest/_static/api.html#tag/containers/operation/ContainerInspectLibpod
+func (c *Client) InspectContainer(ctx context.Context, id string) (*Container, error) {
+	rsp, err := c.get(ctx, fmt.Sprintf("containers/%s", id))
+	if err != nil {
+		return nil, err
+	}
+	defer rsp.Body.Close()
+
+	var cntr Container
+	if err := json.NewDecoder(rsp.Body).Decode(&cntr); err != nil {
+		return nil, trace.Wrap(err, "failed to decode container response")
+	}
+	return &cntr, nil
+}
+
+// InspectPod reads information about the pod with the given ID.
+//
+// https://docs.podman.io/en/latest/_static/api.html#tag/pods/operation/PodInspectLibpod
+func (c *Client) InspectPod(ctx context.Context, id string) (*Pod, error) {
+	rsp, err := c.get(ctx, fmt.Sprintf("pods/%s", id))
+	if err != nil {
+		return nil, err
+	}
+	defer rsp.Body.Close()
+
+	var pod Pod
+	if err := json.NewDecoder(rsp.Body).Decode(&pod); err != nil {
+		return nil, trace.Wrap(err, "failed to decode pod response")
+	}
+	return &pod, nil
+}
+
+func (c *Client) get(ctx context.Context, path string) (*http.Response, error) {
+	url := fmt.Sprintf("http://d/%s/libpod/%s/json", APIVersion, path)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	rsp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	if rsp.StatusCode == http.StatusOK {
+		return rsp, nil
+	}
+
+	if err := rsp.Body.Close(); err != nil {
+		return nil, trace.Wrap(err, "failed to close response body")
+	}
+	switch rsp.StatusCode {
+	case http.StatusNotFound:
+		return nil, trace.NotFound("not found")
+	default:
+		return nil, trace.Errorf("unexpected response from Podman service (status: %d)", rsp.StatusCode)
+	}
+}
+
+// Container details.
+type Container struct {
```

Comment:
For consistency, I tend to declare the `json` tag for all fields if it's needed for one of them.